server/config/config.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/burntsushi/toml"
+	"github.com/pkg/errors"
+)
+
+// HTTPConfig holds Draupnir's HTTP configuration
+type HTTPConfig struct {
+	Port               int    `toml:"port"`
+	InsecurePort       int    `toml:"insecure_port"`
+	TLSCertificatePath string `toml:"tls_certificate"`
+	TLSPrivateKeyPath  string `toml:"tls_private_key"`
+}
+
+// OAuthConfig holds Draupnir's OAuth configuration
+type OAuthConfig struct {
+	RedirectURL  string `toml:"redirect_url"`
+	ClientID     string `toml:"client_id"`
+	ClientSecret string `toml:"client_secret"`
+}
+
+// Config holds all Draupnir configuration
+type Config struct {
+	DatabaseURL            string      `toml:"database_url"`
+	DataPath               string      `toml:"data_path"`
+	Environment            string      `toml:"environment"`
+	SharedSecret           string      `toml:"shared_secret"`
+	TrustedUserEmailDomain string      `toml:"trusted_user_email_domain"`
+	SentryDsn              string      `toml:"sentry_dsn" required:"false"`
+	HTTPConfig             HTTPConfig  `toml:"http"`
+	OAuthConfig            OAuthConfig `toml:"oauth"`
+}
+
+// Load parses and validates the server config file located at `path`
+func Load(path string) (Config, error) {
+	var config Config
+	file, err := os.Open(path)
+	if err != nil {
+		return config, errors.Wrap(err, fmt.Sprintf("No configuration file found at %s", path))
+	}
+
+	_, err = toml.DecodeReader(file, &config)
+	if err != nil {
+		return config, errors.Wrap(err, "Could not parse configuration file")
+	}
+
+	err = validateConfig(config)
+	if err != nil {
+		return config, errors.Wrap(err, "Invalid configuration")
+	}
+
+	return config, nil
+}
+
+func validateConfig(cfg Config) error {
```

Comment:
Maybe [this](https://github.com/go-playground/validator), but understand if you don't want to vendor it.